BUG: false positive in the heuristic for old Accelerate

It was observed in https://github.com/spack/spack/issues/41742 that this error was triggered despite the numpy build not being linked to Accelerate and no other packages being loaded:
```
RuntimeError: Polyfit sanity test emitted a warning, most likely due to using a buggy Accelerate backend.
If you compiled yourself, more information is available at:
https://numpy.org/doc/stable/user/building.html#accelerated-blas-lapack-libraries
Otherwise report this to the vendor that provided NumPy.
RuntimeWarning: invalid value encountered in divide
```

A run of the test suite after disabling the `_mac_os_check` check in `numpy/__init__.py` confirmed that the `numpy` install was basically fine, with the only failures of interest being:
```
FAILED .../numpy/polynomial/tests/test_classes.py::test_bad_conditioned_fit[Polynomial] - AssertionError: assert 'invalid valu...red in divide' == 'The fit may ...y conditioned'
FAILED .../numpy/polynomial/tests/test_classes.py::test_bad_conditioned_fit[Legendre] - AssertionError: assert 'invalid valu...red in divide' == 'The fit may ...y conditioned'
FAILED .../numpy/polynomial/tests/test_classes.py::test_bad_conditioned_fit[Chebyshev] - AssertionError: assert 'invalid valu...red in divide' == 'The fit may ...y conditioned'
FAILED .../numpy/polynomial/tests/test_classes.py::test_bad_conditioned_fit[Laguerre] - AssertionError: assert 'invalid valu...red in divide' == 'The fit may ...y conditioned'
FAILED .../numpy/polynomial/tests/test_classes.py::test_bad_conditioned_fit[Hermite] - AssertionError: assert 'invalid valu...red in divide' == 'The fit may ...y conditioned'
FAILED .../numpy/polynomial/tests/test_classes.py::test_bad_conditioned_fit[HermiteE] - AssertionError: assert 'invalid valu...red in divide' == 'The fit may ...y conditioned'
```

The `numpy.polynomial` tests are separate code from the `_mac_os_check` heuristic, but both do the same thing and rely on a full-rank check after a call to `np.linalg.lstsq`. This seems to be sensitive to the particular build config in the linked Spack issue, which is:
- macOS 12.6.6, arm64
- Apple Clang 13.1.6
- OpenBLAS 0.3.24

This is a problem, because the heuristic failing makes NumPy unimportable with a misleading error message.

Finally, it's worth pointing out that the `lstsq` call uses `np.errstate` to suppress warnings exactly like the one that's making the heuristic fail: 
https://github.com/numpy/numpy/blob/0032ede015c9b06f88cc7f9b07138ce35f4357ae/numpy/linalg/_linalg.py#L2466-L2468

The question is what to do about this. We should probably at least change the hard error to a warning.

--- a/numpy/__init__.py
+++ b/numpy/__init__.py
@@ -461,20 +461,25 @@ else:
             pass
 
     if sys.platform == "darwin":
+        from . import exceptions
         with warnings.catch_warnings(record=True) as w:
             _mac_os_check()
             # Throw runtime error, if the test failed Check for warning and error_message
             if len(w) > 0:
-                error_message = "{}: {}".format(w[-1].category.__name__, str(w[-1].message))
-                msg = (
-                    "Polyfit sanity test emitted a warning, most likely due "
-                    "to using a buggy Accelerate backend."
-                    "\nIf you compiled yourself, more information is available at:"
-                    "\nhttps://numpy.org/devdocs/building/index.html"
-                    "\nOtherwise report this to the vendor "
-                    "that provided NumPy.\n{}\n".format(error_message))
-                raise RuntimeError(msg)
-        del w
+                for _wn in w:
+                    if _wn.category is exceptions.RankWarning:
+                        # Ignore other warnings, they may not be relevant (see gh-25433).
+                        error_message = f"{_wn.category.__name__}: {str(_wn.message)}"
+                        msg = (
+                            "Polyfit sanity test emitted a warning, most likely due "
+                            "to using a buggy Accelerate backend."
+                            "\nIf you compiled yourself, more information is available at:"
+                            "\nhttps://numpy.org/devdocs/building/index.html"
+                            "\nOtherwise report this to the vendor "
+                            "that provided NumPy.\n\n{}\n".format(error_message))
+                        raise RuntimeError(msg)
+                del _wn
+            del w
     del _mac_os_check
 
     def hugepage_setup():

